### PR TITLE
Client operation log ergonomic improvements

### LINKED ISSUE #3028: Make operation meta types more discoverable
I'd probably be good to make it a convention to have the meta type be a associated type of client extension traits.

### LINKED ISSUE #3029: Rename `operation_type` to `operation_module_kind` in `OperationLogEntry`
The only way the property is used right now is to store the module kind.

### LINKED ISSUE #3030: Add paid fee to lightning send operation meta struct

### PR DESCRIPTION
**Pull request by elsirion**
Fixes #3028
Fixes #3029 
Fixes #3030

### CODE REVIEW

File: modules/fedimint-ln-client/src/lib.rs [hunk 1]
```diff
@@ -274,10 +274,28 @@ impl LightningClientExt for Client {
...
+            LightningOutput::Contract(contract) => {
+                let fee_msat = contract
+                    .amount
+                    .msats
+                    .checked_sub(
+                        invoice
+                            .amount_milli_satoshis()
+                            .ok_or(anyhow::anyhow!("MissingInvoiceAmount"))?,
+                    )
+                    .expect("Contract amount should be greater or equal than invoice amount");
```
Comment (douglaz) [inline modules/fedimint-ln-client/src/lib.rs:289]: panic in a client? :thinking:

File: modules/fedimint-ln-client/src/lib.rs [hunk 2]
```diff
@@ -274,10 +274,28 @@ impl LightningClientExt for Client {
...
+        // it/bounds for it
+        let fee = match &output.output {
+            LightningOutput::Contract(contract) => {
+                let fee_msat = contract
+                    .amount
+                    .msats
+                    .checked_sub(
+                        invoice
+                            .amount_milli_satoshis()
+                            .ok_or(anyhow::anyhow!("MissingInvoiceAmount"))?,
```
Comment (douglaz) [inline modules/fedimint-ln-client/src/lib.rs:287]:
```suggestion
                            .context("MissingInvoiceAmount")?,
```

File: modules/fedimint-ln-client/src/lib.rs (hunk 1 above)
Comment (m1sterc001guy) [reply modules/fedimint-ln-client/src/lib.rs:289]: Maybe better to just return an `anyhow::anyhow!` error and prevent the client from paying? The client probably should do some reasonable fee checking before attempting the payment

Comment (elsirion) [reply modules/fedimint-ln-client/src/lib.rs:289]: This happening would indicate a bug in the client logic. Underfunding an outgoing contract doesn't make any sense, the gateway will just reject. So this really cannot happen unless we make a mistake (e.g. forgetting to change this should we ever implement negative fees).

File: modules/fedimint-ln-client/src/lib.rs (hunk 2 above)
Comment (elsirion) [reply modules/fedimint-ln-client/src/lib.rs:287]: TIL: one can use `context` on `Option`s :open_mouth:

**Review by dpc — APPROVED** [review]

**Pull request merged**